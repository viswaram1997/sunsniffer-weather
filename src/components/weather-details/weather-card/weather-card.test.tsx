import { render, fireEvent } from "@testing-library/react";
import WeatherCard from "./index"
import moment from "moment"

let cardDetails = {"date":"2022-03-25","details":{"weather":{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"},"main":{"temp":30.24,"feels_like":35.63,"temp_min":30.24,"temp_max":30.24,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":70,"temp_kf":0}}}
let cardDetails1 = {"date":moment().format('YYYY-MM-DD'),"details":{"weather":{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"},"main":{"temp":30.24,"feels_like":35.63,"temp_min":30.24,"temp_max":30.24,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":70,"temp_kf":0}}}

describe('Weather Card', () => {

    test('should match snapshot', () => {
        const result = render(<WeatherCard
            bestDayToSell=""
            cardDetails={cardDetails}
        />)
        expect(result.baseElement).toMatchSnapshot();
    })

    test('should display details properly', async () => {
        const result = render(<WeatherCard
            bestDayToSell=""
            cardDetails={cardDetails}
        />)
        const label = await result.findByText('30.24 C Â°')
        const weather = await result.findByText('Clouds')
        const humidity = await result.findByText('70%')
        const date = await result.findByText('Fri')

        expect(label).not.toBeNull()
        expect(weather).not.toBeNull()
        expect(humidity).not.toBeNull()
        expect(date).not.toBeNull()
    })
    test('should display best day to sell', async () => {
        const result = render(<WeatherCard
            bestDayToSell="Umberla"
            cardDetails={cardDetails1}
        />)
        const label = await result.findByText('Best day to sell Umberla')
        expect(label).not.toBeNull()
    })
    test('should display date as today', async () => {
        const result = render(<WeatherCard
            bestDayToSell="Umberla"
            cardDetails={cardDetails1}
        />)
        const date = await result.findByText('Today')
        expect(date).not.toBeNull()
    })

});