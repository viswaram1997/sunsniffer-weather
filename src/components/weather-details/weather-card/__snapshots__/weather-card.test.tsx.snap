// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Card should match snapshot 1`] = `
<body>
  <div>
    <div
      class="card-root-wrapper card"
      data-testid="weather-card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-details row"
        >
          <div
            class="col-4"
          >
            Fri
            
          </div>
          <div
            class="col-2"
          >
            70
            %
          </div>
          <div
            class="col-3"
          >
            <img
              src="http://openweathermap.org/img/wn/02d@2x.png"
            />
            <p>
              Clouds
            </p>
          </div>
          <div
            class="col-3"
          >
            30.24
             C Â°
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
